public/app/features/templating/template_srv.ts

```diff
@@ -233,8 +255,10 @@ export class TemplateSrv implements BaseTemplateSrv {
 
       if (scopedVars) {
         const value = this.getVariableValue(variableName, fieldPath, scopedVars);
+        const text = this.getVariableText(variableName, fieldPath, scopedVars);
```

Comment:
this is a bit expensive (if there is a field path), would be nice to only have to do this if format is text. 

Maybe we can have a special logic here for format == text, and then pass that as value to this.formatValue

Reply:
hmm, I can see that we have a cache. Is it expensive to do regardless of it being cached?